app/src/main/java/com/mapbox/maps/testapp/examples/localization/LocalizationSettingsActivity.kt

```diff
@@ -0,0 +1,308 @@
+            Log.d(
+              TAG,
+              "Failed to set language to dataStore. Error : ${expected.error}"
+            )
+          }
+        }
+      }
+    }
+
+    worldViewSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+      override fun onNothingSelected(parent: AdapterView<*>?) {}
+
+      override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+        val selectedTag = worldViewAdapter.getItem(position).tag
+        launch(Dispatchers.IO) {
+          val expected = setWorldView(selectedTag)
+          if (expected.isError) {
+            Log.d(
+              TAG,
+              "Failed to set worldview to dataStore. Error : ${expected.error}"
+            )
+          }
+        }
+      }
+    }
+  }
+
+  override fun onMapLoadError(eventData: MapLoadingErrorEventData) {
+    if (eventData.type == MapLoadErrorType.SOURCE) {
+      val message = eventData.message
+      Toast.makeText(
+        this@LocalizationSettingsActivity, "Error: $message", Toast.LENGTH_SHORT
+      ).show()
+    }
+  }
+
+  private companion object {
+    const val TAG = "LocalizationActivity"
+    const val SOURCE_ID = "source-id"
+  }
+
+  private class LocalizationAdapter(
+    private val baseContext: Context,
+    private val layoutResourceId: Int,
+    val localizationDataList: List<LocalizationData>
+  ) : ArrayAdapter<LocalizationData>(baseContext, layoutResourceId, localizationDataList) {
+
+    override fun getCount(): Int {
+      return localizationDataList.size
+    }
+
+    override fun getItem(position: Int): LocalizationData {
+      return localizationDataList[position]
+    }
+
+    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
+      val view = super.getView(position, convertView, parent)
+      val localizationData = localizationDataList[position]
+      setSpinnerText(view, localizationData)
+      return view
+    }
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+      val view = super.getView(position, convertView, parent)
+      val localizationData = localizationDataList[position]
+      setSpinnerText(view, localizationData)
+      return view
+    }
+
+    private fun setSpinnerText(view: View?, localizationData: LocalizationData) {
+      view?.findViewById<TextView>(R.id.text_view_spinner)?.apply {
+        text = baseContext.resources.getString(
+          R.string.drop_down_text,
+          localizationData.description,
+          localizationData.tag
+        )
+      }
+    }
+  }
+
+  override val coroutineContext: CoroutineContext
+    get() = job + Dispatchers.Default
+
+  /**
+   * Create a localization data that will populate in the drop down view.
+   * language is expected as BCP-47 tag, we extract the language tag from the [Locale] and set
+   * it to SettingsInterface method.
+   */
+  private fun createLocalizationData() {
```

Comment:
why is this a function, couldn't this be done inline and thus making those lists immutable?

Reply:
Maybe move into companion object?